partner_country_lang/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.onchange("country_id")
+    def _onchange_country_id(self):
+        super()._onchange_country_id()
+        if self.country_id.lang:
+            self.lang = self.country_id.lang
```

Comment:
I would have tested if lang is not yet defined as partner lang may be different than the one defined in the country

Reply:
Sorry but I am not sure about the change you suggest.
By default, a language is assigned (default of `lang` field), that's why in the onchange you are forced to change the language.

What approach do you suggest?

Reply:
The idea is to always overwrites the lang, as the country lang has more priority than previously selected lang. If not, imagine this odd case:

- Associate "Spanish" to "Spain" and "French" to "France".
- Create partner.
- Select "France" as country > "French" is selected.
- Now select "Spain" as previous one was a mistake > keeping "French" is incorrect.previously.

Reply:
I guess what Dennis has in mind is to be able to keep exceptions (it can go in the roadmap). For example: Switzerland (french and derman) or Belgium (french and dutch) language depends on factors that sometimes can't be isolated to a country area. To keep those exceptions it would be necessary a language lock mechanism or something like that.

Reply:
But I don't see it as a problem, as the sequence in such case would be:

- Set the country.
- A default language is set (or none).
- You select the correct language.

Reply:
Yes, but sometimes a simple address change can trigger the country onchange, and then the default lang value for that country would be applied.

Reply:
Are you sure about this? I think this was improved in the web client and the onchange is not triggered if current value = previous value.

Reply:
Oh, you're right. The onchange isn't triggered anymore :+1:

Reply:
@rousseldenis is it OK for you then?